cloudigrade/internal/authentication.py

```diff
@@ -35,3 +35,19 @@ class IdentityHeaderAuthenticationInternalCreateUser(IdentityHeaderAuthenticatio
     require_account_number = True
     require_user = False
     create_user = True
+
+
+class IdentityHeaderAuthenticationInternalConcurrent(IdentityHeaderAuthentication):
+    """
+    Authentication class that uses identity header to query/trigger concurrent reports.
+
+    This authentication checks for the identity header and requires the identity
+    to have org_admin enabled.
+    """
+
+    require_org_admin = True
+    require_account_number = True
+    require_user = True
+    create_user = False
+    support_account_number_header = True
+    support_org_admin_header = True
```

Comment:
…and if we collapse the two as I mentioned above, this would be:
```suggestion
    allow_internal_fake_identity_header = True
```